### PR TITLE
read defect info of cppcheck from template stream instead of XML

### PR DESCRIPTION
**Pull request by Min-Ho-Kim**
1. read cppcheck result from template stream
2. improve performance a little bit by creating TranslateUnit once

### CODE REVIEW

File: project/dexter-core/src/java/com/samsung/sec/dexter/core/checker/CheckerConfig.java [hunk 1]
```diff
@@ -22,147 +22,107 @@
...
+    public void checkerListToMap() {
+        if (checkerList == null)
+            return;
+
+        checkerMap = new HashMap<>(checkerList.size());
+        for (Checker checker : checkerList) {
+            checkerMap.put(checker.getCode(), checker);
+        }
+
+        checkerList = null;
```
Comment (Minjung-Baek) [inline project/dexter-core/src/java/com/samsung/sec/dexter/core/checker/CheckerConfig.java]: why do you want to set 'null' for 'checkerList' ?

File: project/dexter-core/src/java/com/samsung/sec/dexter/core/BaseAnalysisEntity.java [hunk 2]
```diff
@@ -86,7 +86,7 @@ protected BaseAnalysisEntity(BaseAnalysisEntity entity) {
      * @return the projectName Logical Project Name (like PLM)
      */
     public String getProjectName() {
-        return projectName;
+        return projectName.intern();
```
Comment (Min-Ho-Kim) [inline project/dexter-core/src/java/com/samsung/sec/dexter/core/BaseAnalysisEntity.java]: That's what I intended

File: project/dexter-core/src/java/com/samsung/sec/dexter/core/checker/CheckerConfig.java (hunk 1 above)
Comment (Min-Ho-Kim) [inline project/dexter-core/src/java/com/samsung/sec/dexter/core/checker/CheckerConfig.java]: to be gc

Comment (Min-Ho-Kim) [inline project/dexter-core/src/java/com/samsung/sec/dexter/core/checker/CheckerConfig.java]: checkerList field will be removed soon

**Pull request merged**